internal/cmd/controller/target/target.go

```diff
@@ -58,22 +58,19 @@ func (t *Target) BundleDeployment() *fleet.BundleDeployment {
 		Spec: t.Deployment.Spec,
 	}
 	bd.Spec.Paused = t.IsPaused()
+
 	for _, bundleTarget := range t.Bundle.Spec.Targets {
-		if bundleTarget.NamespaceLabels != nil {
-			for key, value := range *bundleTarget.NamespaceLabels {
-				(*bd.Spec.Options.NamespaceLabels)[key] = value
-				(*bd.Spec.StagedOptions.NamespaceLabels)[key] = value
-			}
+		for key, value := range bundleTarget.NamespaceLabels {
+			(*bd.Spec.Options.NamespaceLabels)[key] = value
+			(*bd.Spec.StagedOptions.NamespaceLabels)[key] = value
 		}
-	}
-	for _, bundleTargets := range t.Bundle.Spec.Targets {
-		if bundleTargets.NamespaceAnnotations != nil {
-			for key, value := range *bundleTargets.NamespaceAnnotations {
-				(*bd.Spec.Options.NamespaceAnnotations)[key] = value
-				(*bd.Spec.StagedOptions.NamespaceAnnotations)[key] = value
-			}
+
+		for key, value := range bundleTarget.NamespaceAnnotations {
+			(*bd.Spec.Options.NamespaceAnnotations)[key] = value
+			(*bd.Spec.StagedOptions.NamespaceAnnotations)[key] = value
```

Comment:
We should change this for bundledeployments, too.

Reply:
Done :+1: